config/destination_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/config/validator"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/models"
+)
+
+func TestParseDestination(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		in   map[string]string
+		want Destination
+		err  error
+	}{
+		{
+			name: "valid config with only required fields",
+			in: map[string]string{
+				models.ConfigPrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+				models.ConfigClientEmail: "test@test-pubsub.com",
+				models.ConfigProjectID:   "test-pubsub",
+				models.ConfigTopicID:     "test-TOPic.scription~%_1230+",
+			},
+			want: Destination{
+				General: General{
+					PrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+					ClientEmail: "test@test-pubsub.com",
+					ProjectID:   "test-pubsub",
+				},
+				TopicID: "test-TOPic.scription~%_1230+",
+			},
+		},
+		{
+			name: "topic id is too small",
+			in: map[string]string{
+				models.ConfigPrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+				models.ConfigClientEmail: "test@test-pubsub.com",
+				models.ConfigProjectID:   "test-pubsub",
+				models.ConfigTopicID:     "to",
+			},
+			err: validator.InvalidNameErr(models.ConfigTopicID),
+		},
+		{
+			name: "topic id is too big",
```

Comment:
Maybe replace "too small" and "too big" with "too short" and "too long".

Reply:
Deal.